### PR TITLE
Issue 515

### PR DESCRIPTION
**Pull request by AbfrmBlr**

### CODE REVIEW

File: impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java [hunk 1]
```diff
@@ -262,7 +263,27 @@ private static boolean recursiveDelete(File file) {
     }
     return true;
   }
+  
+  private static boolean tryRecursiveDelete(File file) {
+    boolean success = true;
+    for (int i = 0; i < 5; i++) {
+      success = recursiveDelete(file);
```
Comment (alexsnaps) [inline impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java]: Could we add a check on the OS actually being Windows before looping like this?

Comment (alexsnaps) [inline impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java]: Oh... and I won't go into how much I hate this loop, as it's only a temp fix ;)

File: impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java [hunk 2]
```diff
@@ -262,7 +263,27 @@ private static boolean recursiveDelete(File file) {
...
+      if (success) {
+        break;
+      } else {
+        System.gc();
+        System.runFinalization();
 
+        try {
+          Thread.sleep(50);
+        } catch (InterruptedException e) {
+          // do nothing ?
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java]: Probably best to store the interrupted status in a local boolean here and reassert on method exit.

File: impl/src/test/java/org/ehcache/internal/persistence/CacheManagerDestroyRemovesPersistenceTest.java [hunk 3]
```diff
@@ -43,24 +43,36 @@
...
     persistentCacheManager.close();
     Maintainable maintainable = persistentCacheManager.toMaintenance();
     maintainable.destroy();
+    maintainable.close();
 
     assertThat(file.list().length, is(0));
   }
+  
+  @Test
+  public void testDestoryCacheDestroysPersistenceContext() throws URISyntaxException, CachePersistenceException {
```
Comment (chrisdennis) [inline impl/src/test/java/org/ehcache/internal/persistence/CacheManagerDestroyRemovesPersistenceTest.java]: destory --> destroy

**Comment by AbfrmBlr** [comment]
test this please

File: impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java [hunk 4]
```diff
@@ -262,7 +264,38 @@ private static boolean recursiveDelete(File file) {
...
+    } finally {
+      if (interrupted) {
+        Thread.currentThread().interrupt();
+      }
+    }
+    return success;
+ }
+  
+  private static boolean isWindows() {
+    return System.getProperty("os.name").toLowerCase(Locale.ENGLISH).contains("win");
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java]: matches: "Darwin", which I believe some OpenJDK builds might identify OS X as.

File: impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java [hunk 5]
```diff
@@ -262,7 +264,38 @@ private static boolean recursiveDelete(File file) {
...
   }
+  
+  private static boolean tryRecursiveDelete(File file) {
+    boolean interrupted = false;
+    boolean success = true;
+    try {
+      for (int i = 0; i < 5; i++) {
+        success = recursiveDelete(file);
+        if (success || !isWindows()) {
+          break;
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java]: just do return true here.

File: impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java [hunk 6]
```diff
@@ -262,7 +264,38 @@ private static boolean recursiveDelete(File file) {
...
+            interrupted = true;
+          }
+        }
+      }
+    } finally {
+      if (interrupted) {
+        Thread.currentThread().interrupt();
+      }
+    }
+    return success;
```
Comment (chrisdennis) [inline impl/src/main/java/org/ehcache/internal/persistence/DefaultLocalPersistenceService.java]: and return false here.

**Comment by AbfrmBlr** [comment]
test this please

**Pull request merged**